File: src/main/java/net/dean/jraw/fluent/AbstractReference.java
package net.dean.jraw.fluent;

import net.dean.jraw.RedditClient;

/**
 * Provides a basic abstract implementation of {@link Reference}
 */
public abstract class AbstractReference implements Reference {
    protected final RedditClient reddit;

    protected AbstractReference(RedditClient reddit) {
        this.reddit = reddit;
    }

    @Override
    public RedditClient getRedditClient() {
        return reddit;
    }
}

File: lib/src/main/kotlin/net/dean/jraw/references/AbstractReference.kt
package net.dean.jraw.references

import net.dean.jraw.RedditClient

/**
 * Base class for all References.
 *
 * This class provides an internal constructor and default toString(), equals(), and hashCode() override.
 */
abstract class AbstractReference internal constructor(
    /** Used to send HTTP requests and make WebSocket connections */
    protected val reddit: RedditClient
) : Reference {
    /** */
    override fun toString(): String {
        return "AbstractReference(reddit=$reddit)"
    }
}
